python/lsst/ts/mtdome/enums.py

```diff
@@ -20,56 +20,26 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 __all__ = [
-    "IntermediateState",
-    "LlcMotionState",
+    "InternalMotionState",
     "LlcName",
+    "LlcNameDict",
     "OnOff",
     "ResponseCode",
-    "LlcNameDict",
 ]
 
 import enum
 
 from lsst.ts.idl.enums.MTDome import SubSystemId
 
 
-class IntermediateState(enum.IntEnum):
-    BRAKES_DISENGAGED = enum.auto()
-    BRAKES_ENGAGED = enum.auto()
-    DEFLATED = enum.auto()
-    DEFLATING = enum.auto()
-    DISABLING_MOTOR_POWER = enum.auto()
-    DISENGAGING_BRAKES = enum.auto()
-    ENABLING_MOTOR_POWER = enum.auto()
-    ENGAGING_BRAKES = enum.auto()
-    GO_DEGRADED = enum.auto()
-    GO_NORMAL = enum.auto()
-    GO_STATIONARY = enum.auto()
-    INFLATED = enum.auto()
-    INFLATING = enum.auto()
-    LP_DISENGAGED = enum.auto()
-    LP_DISENGAGING = enum.auto()
-    LP_ENGAGED = enum.auto()
-    LP_ENGAGING = enum.auto()
-    MOTOR_COOLING_OFF = enum.auto()
-    MOTOR_COOLING_ON = enum.auto()
-    MOTOR_POWER_OFF = enum.auto()
-    MOTOR_POWER_ON = enum.auto()
-    STARTING_MOTOR_COOLING = enum.auto()
-    STOPPING_MOTOR_COOLING = enum.auto()
-
-
-class LlcMotionState(enum.IntEnum):
-    """Motion states."""
-
-    CLOSED = enum.auto()
-    CRAWLING = enum.auto()
-    ERROR = enum.auto()
-    MOVING = enum.auto()
-    OPEN = enum.auto()
-    PARKED = enum.auto()
+class InternalMotionState(enum.IntEnum):
+    """Internal Motion states.
+
+    These get translated into IDL MotionState instances by the CSC.
+    """
+
+    DISABLED = enum.auto()
```

Comment:
I am quite uneasy about this enum because it has no translation information and because it sounds like an enum of internal motion states (which might be a very useful thing to have for your mock code, if it was complete).

Please consider saving the translation info as a dict of internal state name: external state name. That way the translation is obvious and also it is clear why it's short.

Please also consider programmatically generating a full InternalMotionState enum class (or using the old name LlcMotionState) for use in your mock code. This should be fairly easy to generate from the MotionState enum and the translation dict (probably the inverse of this dict, which is also easy to make on the fly). In case you are not familiar with the enum functional API, here is a trivial example (which happens to have values != names, but in your case I think you want them to match):
```
name_value_list = (("ANAME", "AVALUE"), ("BNAME", "BVALUE"))
Example = enum.Enum("Example", name_value_list)
```

Reply:
I added a `motion_state_translations` dict that contains the translation info that now is used by the CSC. This has simplified the code there. Thanks for the suggestion.

As for the programmatically generated enum, this works in the code but MyPy goes completely berserk over it. I prefer leaving the code as is over adding `# type: ignore` comments everywhere.